--- a/build/config/mac/mac_sdk.gni
+++ b/build/config/mac/mac_sdk.gni
@@ -18,14 +18,14 @@ declare_args() {
   # `mac_min_system_version` when dropping support for older macOSes but where
   # additional code changes are required to be compliant with the availability
   # rules.
-  mac_deployment_target = "10.13"
+  mac_deployment_target = "10.14"
 
   # The value of the `LSMinimumSystemVersion` in `Info.plist` files. This
   # partially controls the minimum supported version of macOS for Chromium by
   # affecting the `Info.plist`. This may differ from `mac_deployment_target`
   # when dropping support for older macOSes. This must be greater than or equal
   # to the `mac_deployment_target` version.
-  mac_min_system_version = "10.13"
+  mac_min_system_version = "10.14"
 
   # Path to a specific version of the Mac SDK, not including a slash at the end.
   # If empty, the path to the lowest version greater than or equal to
