--- a/ui/events/keycodes/keyboard_code_conversion_mac.mm
+++ b/ui/events/keycodes/keyboard_code_conversion_mac.mm
@@ -48,6 +48,11 @@
   return base::IsValidCodepoint(c) && !base::IsUnicodeControl(c);
 }
 
+#if (__MAC_OS_X_VERSION_MAX_ALLOWED < 150000)
+constexpr int kVK_RightCommand = 0x36;
+constexpr int kVK_ContextualMenu = 0x6E;
+#endif
+
 bool IsKeypadOrNumericKeyEvent(NSEvent* event) {
   // Check that this is the type of event that has a keyCode.
   switch (event.type) {
