--- a/tools/mac/dump-static-initializers.py
+++ b/tools/mac/dump-static-initializers.py
@@ -39,7 +39,7 @@ def ParseDsymutil(binary):
   static initializers.
   """
 
-  child = subprocess.Popen(['tools/clang/dsymutil/bin/dsymutil', '-s', binary],
+  child = subprocess.Popen(['/usr/bin/dsymutil', '-s', binary],
      stdout=subprocess.PIPE)
   for line in child.stdout:
     file_match = dsymutil_file_re.search(line)
--- a/build/toolchain/mac/BUILD.gn
+++ b/build/toolchain/mac/BUILD.gn
@@ -188,8 +188,7 @@ template("mac_toolchain") {
       dsym_switch = " -Wcrl,dsym,{{root_out_dir}} "
       if (is_mac) {
         dsym_switch += "-Wcrl,dsymutilpath," +
-                          rebase_path("//tools/clang/dsymutil/bin/dsymutil",
-                                      root_build_dir) + " "
+                          "/usr/bin/dsymutil "
       }
 
       dsym_output_dir =
