description: avoid building the swiftshader library
author: Michael Gilbert <mgilbert@debian.org>

--- a/BUILD.gn
+++ b/BUILD.gn
@@ -446,7 +446,7 @@ group("gn_all") {
     ]
   }

-  if ((is_win || is_mac || is_linux || is_chromeos || is_fuchsia) &&
+  if (false &&
       (target_cpu == "x86" || target_cpu == "x64")) {
     deps += [ "//third_party/swiftshader" ]
   }
--- a/ui/gl/features.gni
+++ b/bui/gl/features.gni
@@ -9,5 +9,5 @@

   # Should EGL support be compiled. Can be overriden to test during bring up
   # of EGL support on other platforms
-  use_egl = is_win || is_android || is_linux || is_fuchsia || is_mac
+  use_egl = is_win || is_android || is_linux || is_fuchsia
 }
