--- a/chrome/browser/download/bubble/download_bubble_update_service.cc
+++ b/chrome/browser/download/bubble/download_bubble_update_service.cc
@@ -87,6 +87,7 @@ ItemSortKey GetSortKey(const Item& item)
 // Helper to get an iterator to the last element in the cache. The cache
 // must not be empty.
 template <typename Item>
+typename
 SortedItems<Item>::iterator GetLastIter(SortedItems<Item>& cache) {
   CHECK(!cache.empty());
   auto it = cache.end();
@@ -789,8 +790,10 @@ bool DownloadBubbleUpdateService::Remove
 }
 
 template <typename Id, typename Item>
+typename
 SortedItems<Item>::iterator
 DownloadBubbleUpdateService::RemoveItemFromCacheByIter(
+    typename
     SortedItems<Item>::iterator iter,
     SortedItems<Item>& cache,
     IterMap<Id, Item>& iter_map) {
--- a/chrome/browser/download/bubble/download_bubble_update_service.h
+++ b/chrome/browser/download/bubble/download_bubble_update_service.h
@@ -208,7 +208,9 @@ class DownloadBubbleUpdateService
 
   // Removes item if we already have the iterator to it. Returns next iterator.
   template <typename Id, typename Item>
+  typename
   SortedItems<Item>::iterator RemoveItemFromCacheByIter(
+      typename
       SortedItems<Item>::iterator iter,
       SortedItems<Item>& cache,
       IterMap<Id, Item>& iter_map);
