description: fix mojo layout test build error
author: Michael Gilbert <mgilbert@debian.org>

--- a/content/shell/BUILD.gn
+++ b/content/shell/BUILD.gn
@@ -239,7 +239,6 @@ static_library("content_shell_lib") {
     "//content/app/resources",
     "//content/public/common",
     "//content/public/common:service_names",
-    "//content/test:content_test_mojo_bindings",
     "//content/test:test_support",
     "//device/bluetooth",
     "//media",
