description: fix mojo layout test build error
author: Michael Gilbert <mgilbert@debian.org>

--- a/content/shell/BUILD.gn
+++ b/content/shell/BUILD.gn
@@ -185,7 +185,6 @@ if (support_web_tests) {
       "//content/browser:for_content_tests",  # For non-component builds.
       "//content/public/browser",  # For component builds.
       "//content/test:blink_test_browser_support",
-      "//content/test:mojo_web_test_bindings",
       "//content/test:test_support",
       "//device/bluetooth:fake_bluetooth",
       "//device/bluetooth:mocks",
@@ -505,7 +504,6 @@ static_library("content_shell_lib") {
     "//content/app/resources",
     "//content/public/common",
     "//content/public/common:service_names",
-    "//content/test:content_test_mojo_bindings",
     "//content/test:test_support",
     "//device/bluetooth",
     "//media",
