description: fix mojo layout test build error
author: Michael Gilbert <mgilbert@debian.org>

--- a/content/shell/BUILD.gn
+++ b/content/shell/BUILD.gn
@@ -429,8 +429,6 @@ static_library("content_shell_lib") {
     "//content/public/common",
     "//content/public/common:service_names",
     "//content/test:blink_test_browser_support",
-    "//content/test:content_test_mojo_bindings",
-    "//content/test:mojo_web_test_bindings",
     "//content/test:test_support",
     "//content/test:web_test_support_browser",
     "//device/bluetooth",
